### ISSUE TITLE
Make axes dynamic with dropdown but fixed with slider

### ISSUE THREAD

**Issue by helgaai** [altair-viz/altair#1668]
Hey! 

I'm creating a plot where I both have a slider determining the hour and a drop-down determining the category I want to look at. 

The code looks like this:

```
import altair as alt

categories = list(dfHourMany['Item_CategoryDescription'].unique())

slider = alt.binding_range(min=0, max=23, step=1)
select_hour = alt.selection_single(name='hour', fields=['hour'],
                                   bind=slider, init={'hour': 17})
# A dropdown filter
cat_dropdown = alt.binding_select(options=categories)
cat_select = alt.selection_single(fields=['Item_CategoryDescription'], bind=cat_dropdown, name="Category")

alt.Chart(dfHourMany).mark_point().encode(
    x=alt.X('TotalSaleMargin:Q', title='Heildarframlegð'),
    y=alt.Y('QtySales:Q', title='Magn selt eftir klukkutíma'),
    #color='Item_GroupDescription:N',
    tooltip="Item_GroupDescription:N"
    ).add_selection(
        select_hour
    ).transform_filter(
        select_hour
    ).add_selection(
        cat_select
    ).transform_filter(
        cat_select
    ).properties(title="Vöruflokkar")
```

and the resulting plot is like this:

![image](https://user-images.githubusercontent.com/54354092/63421498-33ceb000-c3f8-11e9-8107-0a8125d82d46.png)

Since different categories in the dropdown return vastly different magnitudes of values, I want the axes to change for each category chosen there, for example:

![image](https://user-images.githubusercontent.com/54354092/63421776-b5264280-c3f8-11e9-90d2-a81ab32f7cce.png)

However, when I drag the slider to change the hour, the graph interactively changes so I don't really see the change, there's always one item at the far right, which I don't have a feeling for how large is in relation to other hours of the day.

![image](https://user-images.githubusercontent.com/54354092/63422020-1cdc8d80-c3f9-11e9-9bbf-fc6baa354acd.png)

**Is it possible to fix the slider but keep the dropdown dynamic? I can't find any information on the problem.**

Thanks!

**Comment by jakevdp** [comment]
I'm having trouble understanding exactly what behavior you're seeing and how it differs from the behavior you want to see, because you did not provide a complete code snippet to reproduce your chart. I also don't know what you mean by "fix the slider but keep the dropdown dynamic".

But perhaps you could be more specific about how you combine the selections; e.g. do something like:
```python
alt.Chart(...).add_selection(
    select_hour, cat_select
).transform_filter(
    select_hour & cat_select
)
```

**Comment by helgaai** [comment]
Thanks! It's not a public dataset so unfortunately I can't provide a complete code snippet. :) The suggestion didn't change anything.

I'm sorry about not being clear, 

I want to choose a category in the dropdown list and then get the axes as a minimum to maximum of the chosen category. 

I don't want the axes to change as I slide to get different hours.

So I would choose "Rekstrarvörur" and get x-axis from -200,000 (because that's the lowest margin of the category as a whole) to 25,000,000 (which is the highest margin). Now it's at this at 2pm but only shows the x-axis for 0-100,000 in the middle of the night. I want to be able to see how the margin changes during the day but it's hard when the axes keep changing, then it's not very clear visually.

Then I want to choose another category in the dropdown list and get other fixed axes that don't change with the slider, but represent the lowest and highest margin of the category.

Basically: 
- choose a dropdown category, get axes that represent the min and max of this category. The bubbles in the plot show subgroups of the categories. I have those values mapped to each row in my pandas dataframe (the min and max).
- slide to see different hours of the day without the axes changing.

Is it still unclear? I could try to reproduce the same with a public dataset from vega. 

This is a very straightforward thought but apparently really hard for me to get across! Thanks!

**Comment by Alcampopiano** [comment]
@helgaai You wouldn't necessarily have to share the exact dataset. Rather, you could construct a small DataFrame of fake data and use that. This way, others can simply copy and paste your code.

**Comment by helgaai** [comment]
This should be reproducible. :)

```
data = [[0, 'construction', 'paint', 1, 9300],
        [0, 'construction', 'paper', 20, 15000],
        [1, 'construction', 'paint', 7, 70000],
        [1, 'construction', 'paper', 200, 150000],
        [0, 'home', 'bicycles', 2, 35200],
        [0, 'home', 'batteries', 3, 900],
        [1, 'home', 'bicycles', 6, 600000],
        [1, 'home', 'batteries', 68, 10000]
       ]
dfFake = pd.DataFrame(data, columns=['hour', 'category', 'group', 'qtysales', 'margin'])

import altair as alt

categories = list(dfFake['category'].unique())

slider = alt.binding_range(min=0, max=1, step=1)
select_hour = alt.selection_single(name='hour', fields=['hour'],
                                   bind=slider, init={'hour': 0})
# A dropdown filter
cat_dropdown = alt.binding_select(options=categories)
cat_select = alt.selection_single(fields=['category'], bind=cat_dropdown, name="Category")

alt.Chart(dfFake).mark_point().encode(
    x=alt.X('margin:Q', title='Margin'),# scale=alt.Scale(domain=[dfHourMany['minMargin'], dfHourMany['maxMargin']])),
    y=alt.Y('qtysales:Q', title='Sales'),#, scale=alt.Scale(domain=[0, maxManySala]))
    color='group:N',
    tooltip='group:N'
    ).add_selection(
        select_hour
    ).transform_filter(
        select_hour
    ).add_selection(
        cat_select
    ).transform_filter(
        cat_select
    ).properties(title="Product groups")
```
